src/api/osaamiset.ts

```diff
@@ -0,0 +1,86 @@
+import { client } from './client';
+import { components } from './schema';
+
+type OsaaminenDto = components['schemas']['OsaaminenDto'];
+const cache = new Map<string, OsaaminenDto>();
+const maxQueryLength = 4096;
+const queryParamOverhead = 5; // uri=...&
+
+export const osaamiset = {
+  /**
+   * Combines objects referencing osaamiset with osaamiset.
+   *
+   * @param objects Collection of objects to combine with osaamiset.
+   * @param key Property accessor for the osaaminen uri.
+   * @param combiner Function to combine the object with the osaaminen.
+   * @param signal Abort signal.
+   * @returns An array of objects combined with osaamiset.
+   */
```

Comment:
Would be good to explain here that `combine()` makes (one or multiple) API calls to fetch the data. Otherwise one might assume that it combines some data that already exists.

(would a different name make sense, `complete()` maybe?)